rfcs/20210119-determinism.md

```diff
@@ -81,3 +80,26 @@ As part of the implementation, we will review all ops to make a determination of
 `tf.image.sample_distorted_bounding_box` has been observed to behave nondeterministically unless you set its seed parameter, even if you call tf.random.set_seed. We will review this Op as part the change. Another case that needs review is "pulling a random number from a PRNG before its state has been initialized".
 
 Given the large number of ops involved, there is a chance that we might omit raising an error for a nondeterministic Op.
+
+## Discussion
+
+This section describes some topics that were discussed during the design review on February 4, 2021. The RFC was not yet approved due to the concerns described in the next section.
+
+### CPU Support
+
+If we implement determinism, we do not want to leave it half completed due to a lack of time to work on it. As CPU support may be time consuming, we could start by having the API only supported for GPU and TPU users. However, GPU models may still use some CPU ops. For example, all `tf.data` ops only support the CPU, and most int32 ops run on the CPU, [even for GPU kernels](https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/kernels/identity_op.cc;l=94;drc=3cbb50768909c585d33e99ba10172d1c44c04d6f). To support GPU models, we must also support determinism in these CPU ops. To avoid having to modify most CPU ops to raise a determinism error, we can have an allowlist of allowed CPU ops which are deterministic.
```

Comment:
Most CPU ops are deterministic already. There are very few examples that I have ever found of ops functioning nondeterministically on CPU.

Also, integer ops are usually effectively deterministic (even on GPU) because integer addition is truly associative. If you want to be pedantic about it, some GPU ops with integer operands may have nondeterministic handling of overflow and underflow conditions because they are sometimes not commutative, but this is not what users are usually concerned about.